### PR TITLE
feat: GTFS related endpoints

### PR DESCRIPTION
**Pull request by jujiang526**
**Summary:**
[Linked Issue](https://github.com/MobilityData/mobility-feed-api/issues/33)

Implemented the following endpoints without supporting filtering or sorting

[Get GTFS feeds](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/feeds/getGtfsFeeds)
[Get GTFS RT feeds](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/feeds/getGtfsRtFeeds)
[Get GTFS feed by ID](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/feeds/getGtfsFeed)
[Get GTFS RT feed by ID](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/feeds/getGtfsRtFeed)
[Get GTFS feed datasets for a given feed ID](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/feeds/getGtfsFeedDatasets)
[Get GTFS datasets by dataset ID](https://mobilitydata.github.io/mobility-feed-api/SwaggerUI/index.html#/datasets/getDatasetGtfs)

**Expected behavior:** 

Tested all those end points. The performance is good. Added a few unit tests regarding bounding filtering and database join query.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [X] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [X] Linked all relevant issues
- [X] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by aronza** [comment]
Hi @jujiang526 ,

Thank you for all this! I've refactored it further to reduce code duplication.

Also I saw that you were mapping database feed or dataset IDs directly to API IDs. I believe @jcpitre clarified in an earlier meeting that they want us to use stable_id when accepting or returning IDs in API. I've made the changes in the code and verified in my local Postman it works as expected.

Though the tests you've added were using these database IDs that are auto-generated in https://github.com/MobilityData/mobility-feed-api/blob/02f2b6f0b2a4dddac51f7e02de220c965e9004f4/api/src/scripts/populate_db.py#L123C1-L123C1 in our locals, so these tests won't pass anywhere else. Do you know which dataset stable ID did you use for these values?

**Comment by indraneel** [comment]
🚨 Do not merge this PR into main, **until** #94 is reviewed/merged into this branch

**Comment by aronza** [comment]
> 🚨 Do not merge this PR into main, **until** #94 is reviewed/merged into this branch

merged and resolved the conflicts from main

**Comment by jcpitre** [comment]
Added https://github.com/MobilityData/mobility-feed-api/issues/96 to track filtering on dates for feeds.

File: api/src/database/database.py
Comment (jcpitre) [inline api/src/database/database.py:133]: If I am not mistaken, groupby needs a list sorted by the key (in our case stable_id).
I just want to make sure it's sorted. 
If it is, I suggest you put a comment like: "Results are sorted by stable_id since we used the ORDER BY in such and such place", or whatever mean was used to sort.

Comment (indraneel) [reply api/src/database/database.py:133]: @aronza to take this

File: docs/DatabaseCatalogAPI.yaml
Comment (jcpitre) [inline docs/DatabaseCatalogAPI.yaml:87]: There's also filtering by producer_URL according to https://github.com/MobilityData/mobility-feed-api/issues/33
It could be delegated to a new issue.

**Comment by jujiang526** [comment]
@aronza - Can you please try to run the unit test in your environment? I believe I have fixed it. I also fixed a small issue with returning duplicated records from the joined table.

**Comment by aronza** [comment]
> @aronza - Can you please try to run the unit test in your environment? I believe I have fixed it. I also fixed a small issue with returning duplicated records from the joined table.

They pass on my local too @jujiang526 . Though our changes to database class broke @goorui 's unit tests, so I refactored these tests to work with the latest changes

File: api/src/database/database.py
Comment (aronza) [reply api/src/database/database.py:133]: Thanks for pointing this out @jcpitre , I've added order by clause to queries using groupby

File: docs/DatabaseCatalogAPI.yaml
Comment (aronza) [reply docs/DatabaseCatalogAPI.yaml:88]: added to all feeds endpoints

File: api/src/database/database.py
Comment (aronza) [reply api/src/database/database.py:30]: @jujiang526 , I believe you've added this. Any thoughts?

**Review by jcpitre — APPROVED** [review]

**Pull request merged**
